mongoDbTest.js

```diff
@@ -0,0 +1,281 @@
+	const name = req.body.name;
+	let errMsg = null;
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+	try {
+		const results = await db.collection('testCollection').deleteOne({email: email, name: name});
+		console.log(results);
+		if(results.deletedCount === 0) {
+			errMsg = "Could not find person to delete."
+			throw new Error(errMsg);
+		}
+		// Send success response
+		res.send({ status: "success", msg: "successfully deleted person." });
+	} catch(e) {
+		console.log(e);
+		//send fail response
+		res.send({ status: "fail", msg: errMsg ? errMsg : "could not delete person." });
+	}
+})
+
+app.post("/login", async (req, res) => {
+	const email = req.body.email;
+	const pwd = req.body.password;
+	let errMsg = null;
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+	try {
+		const results = await db.collection('users').find({emailAddress: email}).toArray();
+		if(results.length === 0) {
+			// Could not find user
+			res.status(401).send("Invalid email");
+		} else if (results.length > 1) {
+			// Many users with same email
+			// TODO: This is bad and should let someone know
+			errMsg = "Found multiple users with that email address.";
+			throw new Error();
+		} else {
+			// found user. validate password
+			bcrypt.compare(pwd, results[0].pwd, function(err, result) {
+				if(result) {
+					// Password matches, create session
+					req.session.loggedIn = true;
+					req.session.save(err => {
+						if(err)
+							console.log(err);
+					})
+					res.send("Logged in.");
+				} else {
+					// Password does not match
+					res.status(401).send("Invalid password");
+				}
+			});
+		}
+	} catch(e) {
+		console.log(e);
+		if(errMsg) {
+			res.status(403).send(errMsg);
+		} else {
+			res.status(500).send("There was most likely a problem with the server");
+		}
+	}
+})
+
+app.post("/logout", (req, res) => {
+	// TODO: Not sure how to do the error handling here
+	if(req.session) {
+		req.session.destroy(err => {
+			if(err) {
+				// TODO: Should this be a 400 or 500?
+				res.status(400).send("Unable to logout");
+			} else {
+				res.redirect(200, "/");
+			}
+		})
+	} 
+})
+
+app.post("/signup", async (req, res) => {
+	// TODO: Should probably sanitize some of these before sticking them in the database
+	const name = req.body.name;
+	const emailAddress = req.body.email;
+	const pwd = req.body.password;
+	const avatarURL = req.body.avatarURL;
+	const date = new Date().toISOString();
+	const achievements = req.body.achievements;
+	const admin = false;
+	const kit = {
+		description: "",
+		order: [],
+		items: []
+	};
+
+	// Set response header regardless of success/failure
+	res.setHeader("Content-Type", "application/json");
+
+	try {
+		await db.collection('users')
+			.insertOne({name, emailAddress, pwd, avatarURL, date, achievements, admin, kit});
+
+		// TODO: This should redirect to the login page when thats done		
```

Comment:
Resolve this `TODO` if it's a quick change/doesn't require further work/won't be clobbered by future work, otherwise just make an issue for this and do the redirect later.